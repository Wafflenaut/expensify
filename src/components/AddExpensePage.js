import React from 'react';

const AddExpensePage = () => (
	<div>Add Expense Placeholder</div>
	
);

export default AddExpensePage;