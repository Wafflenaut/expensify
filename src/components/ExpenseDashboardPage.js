import React from 'react';

const ExpenseDashboardPage = () => (
	<div>Dashboard Placeholder</div>
	
);

export default ExpenseDashboardPage;