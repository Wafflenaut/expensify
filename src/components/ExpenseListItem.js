import React from 'react';
import moment from 'moment';
import {Link} from 'react-router-dom';


const ExpenseListItem = ({ dispatch, id, description, amount, createdAt}) => (
	<div>
		<h3><Link to={`/edit/${id}`}>{description}</Link></h3>
		<p>{amount} - {moment(createdAt).format('MM-DD-YYYY')}</p>
	</div>
);

export default ExpenseListItem;