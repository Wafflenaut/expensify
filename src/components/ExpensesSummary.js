import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = (props) => (
	<div>
		{
			props.expenses.length === 1 ? (
				<h5>Viewing 1 expense totalling {numeral(props.total / 100).format('$0,0.00')}.</h5>
			) : (
				<h5>Viewing {props.expenses.length} expenses totalling {numeral(props.total / 100).format('$0,0.00')}.</h5>
			)
		}
	</div>
);

const mapStateToProps = (state) => {
	return {
		expenses: selectExpenses(state.expenses, state.filters),
		total: selectExpensesTotal(selectExpenses(state.expenses, state.filters))
	};
};

export default connect(mapStateToProps)(ExpensesSummary);