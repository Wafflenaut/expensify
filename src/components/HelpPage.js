import React from 'react';

const HelpPage = () => (
	<div>Help Page Placeholder</div>
	
);

export default HelpPage;