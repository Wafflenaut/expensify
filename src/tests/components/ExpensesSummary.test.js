import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with 0 or multiple expenses', () => {
	const wrapper = shallow(<ExpensesSummary expenses={expenses} total={500} />);
	expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with single expense', () => {
	const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} total={600}/>);
	expect(wrapper).toMatchSnapshot();
});