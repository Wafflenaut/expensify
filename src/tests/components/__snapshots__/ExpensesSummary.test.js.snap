// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 0 or multiple expenses 1`] = `
<div>
  <h5>
    Viewing 
    3
     expenses totalling 
    $5.00
    .
  </h5>
</div>
`;

exports[`should render ExpensesSummary with single expense 1`] = `
<div>
  <h5>
    Viewing 1 expense totalling 
    $6.00
    .
  </h5>
</div>
`;
